### PR TITLE
feat: add Request::getBestAcceptEncoding() function

### PR DESCRIPTION
**Pull request by Tachi107**
The new getBestAcceptEncoding() function makes using the new compression support introduced in commit b0b7d3076dcdba48935cfe14049b76809d0cd256 way easier, as it reports to the caller the most sensible content encoding to apply to the response, falling back to the Identity encoding (i.e. no actual compression) in case no encoding is supported by both the server and the client.

Fixes https://github.com/pistacheio/pistache/issues/1148

### CODE REVIEW

**Review by kiplingw — APPROVED** [review]
Solid work @Tachi107. LGTM. @dennisjenkins75, have a look over. If you're fine with it lets merge.

**Comment by Tachi107** [comment]
Added some more tests and fixed a corner case discovered with the newly added tests. Should be ready now :)

**Comment by kiplingw** [comment]
@Tachi107, if your PR passes in CI and @dennisjenkins75 has had a chance to review, I'm totally fine with it. Only thing I'd suggest changing is to bump the patch version.

**Comment by Tachi107** [comment]
> Only thing I'd suggest changing is to bump the patch version.

Done

**Comment by Tachi107** [comment]
Debian testing CI runs are failing the Coverage step because of Debian bug <https://bugs.debian.org/1053252>, it's unrelated to this patch set.

File: include/pistache/http.h [hunk 1]
```diff
@@ -217,6 +217,16 @@ namespace Pistache
 
             std::chrono::milliseconds timeout() const;
 
+            /*
+             * Returns the "best" encoding to use to encode (tipically compress)
```
Comment (dennisjenkins75) [inline include/pistache/http.h:221]: "typically"

**Review by dennisjenkins75 — CHANGES_REQUESTED** [review]

File: include/pistache/http.h (hunk 1 above)
Comment (Tachi107) [reply include/pistache/http.h]: Thanks, fixed!

**Comment by Tachi107** [comment]
Thanks for the review and merge :)
